app/src/main/java/ahmaabdo/readify/rss/parser/RssAtomParser.java

```diff
@@ -376,7 +376,8 @@ public void endElement(String uri, String localName, String qName) throws SAXExc
                 mEntryUpdateDate = mPreviousEntryUpdateDate;
             }
 
-            if (mTitle != null && (mEntryDate == null || (mEntryDate.after(mRealLastUpdateDate) && mEntryDate.after(mKeepDateBorder)))) {
+            // this condition check breaks the possibility of changing keep time for individual feeds
+            if (mTitle != null && (mEntryDate == null || (/*mEntryDate.after(mRealLastUpdateDate) &&*/ mEntryDate.after(mKeepDateBorder)))) {
```

Comment:
As explained above, the commented part leads to bad functionality if the user changed the keep time for the feed to be larger after the last syncing process. When leaving it, it won't load older entries.

Another option could be to re-sync automatically after changing the feed's settings without checking this part and then leaving the check to prevent overhead (see line [484](files#diff-c81a604f740d229abc19da0e965b2f65R484)).